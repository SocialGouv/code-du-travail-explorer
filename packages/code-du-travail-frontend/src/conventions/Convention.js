import React from "react";
import ConventionInfos from "./ConventionInfos";
import ConventionTexte from "./ConventionTexte";
import ConventionExplainer from "./ConventionExplainer";
import ConventionListeTextes from "./ConventionListeTextes";
import { Tabs } from "@cdt/ui";

const Convention = ({ convention, conteneur }) => (
  <React.Fragment>
    <ConventionExplainer />
    <Tabs
      data={[
        {
          tab: "Informations",
          panel: (
            <ConventionInfos convention={convention} conteneur={conteneur} />
          ),
          key: "infos"
        },
        {
          tab: "Texte de base",
          panel: <ConventionTexte id={conteneur.texte_de_base} />,
          key: "texte-de-base"
        },
        {
          tab: "Texte attach√©s",
          panel: (
            <ConventionListeTextes
              conteneur={conteneur}
              typeTextes={"attaches"}
            />
          ),
          key: "attaches"
        },
        {
          tab: "Texte salaires",
          panel: (
            <ConventionListeTextes
              conteneur={conteneur}
              typeTextes={"salaires"}
            />
          ),
          key: "salaires"
        },
        {
          tab: "Textes d'extensions",
          panel: (
            <ConventionListeTextes
              conteneur={conteneur}
              typeTextes={"extensions"}
            />
          ),
          key: "extensions"
        }
      ]}
    />
  </React.Fragment>
);

export default Convention;
