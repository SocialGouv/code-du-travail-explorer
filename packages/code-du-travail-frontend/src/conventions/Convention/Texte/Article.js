import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { theme } from "@cdt/ui/";

const ETAT_LABEL = {
  MODIFIE: "Modifie",
  DENONCE: "Dénonce",
  VIGUEUR_ETEN: "Article étendu",
  VIGUEUR: "",
  PERIME: "Périmé",
  ABROGE: "Abrogé",
  VIGUEUR_NON_ETEN: "Article non étendu",
  REMPLACE: "Remplace"
};

const Article = ({ id, etat, surtitre, content }) => {
  return (
    <>
      <Status>{ETAT_LABEL[etat]}</Status>
      {surtitre && <h5 id={id}>{surtitre}</h5>}
      <ArticleContent dangerouslySetInnerHTML={{ __html: content }} />
    </>
  );
};

Article.propTypes = {
  id: PropTypes.string.isRequired,
  surtitre: PropTypes.string,
  etat: PropTypes.oneOf(Object.keys(ETAT_LABEL)),
  content: PropTypes.string.isRequired
};

export default Article;

const { spacing } = theme;

const Status = styled.div`
  margin-bottom: ${spacing.small};
  font-style: italic;
  font-weight: bold;
  font-size: 1em;
`;

const ArticleContent = styled.div`
  margin-bottom: ${spacing.small};
`;
