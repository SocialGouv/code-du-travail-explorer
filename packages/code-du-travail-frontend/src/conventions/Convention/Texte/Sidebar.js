import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { Section, theme, Wrapper } from "@socialgouv/react-ui";
import SidebarItem from "./SidebarItem";

const Sidebar = ({ node }) => {
  const { children = [] } = node;

  const childSections = children.filter(
    ({ type, data: { surtitre } }) =>
      type === "section" || (type === "article" && surtitre)
  );
  // We want to hide the sidebar if there is less than 2 items
  if (childSections.length < 2) {
    return null;
  }

  return (
    <StyledSection>
      <StyledWrapper variant="dark">
        <ol>
          {children.map((childNode, index) => (
            <SidebarItem key={index} node={childNode} />
          ))}
        </ol>
      </StyledWrapper>
    </StyledSection>
  );
};

SidebarItem.propTypes = {
  node: PropTypes.shape({
    children: PropTypes.array
  }).isRequired
};

export default Sidebar;

const { breakpoints, spacings } = theme;

const StyledSection = styled(Section)`
  @media (min-width: ${breakpoints.tablet}) {
    position: sticky;
    top: 10rem;
    flex: 0 1 35%;
    max-height: 100vh;
    padding-right: ${spacings.medium};
  }
`;

const StyledWrapper = styled(Wrapper)`
  max-height: 100%;
  overflow: auto;
`;
