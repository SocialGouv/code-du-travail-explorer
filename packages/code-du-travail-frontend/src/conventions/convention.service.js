import fetch from "isomorphic-unfetch";
import getConfig from "next/config";
import memoizee from "memoizee";
import debounce from "debounce-promise";

const {
  publicRuntimeConfig: { API_URL }
} = getConfig();

const formatCCn = ({ idcc: num, id, slug, title, shortTitle }) => ({
  id,
  slug,
  title,
  shortTitle,
  num
});

// memoize search results
const apiIdcc = memoizee(
  query => {
    const url = `${API_URL}/idcc?q=${encodeURIComponent(query)}`;

    return fetch(url).then(response => {
      if (response.ok) {
        return response
          .json()
          .then(results =>
            results.hits.hits.map(({ _source }) => formatCCn(_source))
          );
      }
      throw new Error("Un probl√®me est survenu.");
    });
  },
  { promise: true }
);

// debounced versions

const searchConvention = debounce(apiIdcc, 300);

export { searchConvention };
