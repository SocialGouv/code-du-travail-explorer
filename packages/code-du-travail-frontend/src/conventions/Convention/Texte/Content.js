import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { theme } from "@cdt/ui-old/";
import { sortByIntOrdre } from "../../utils";
import Article from "./Article";

// Beware, this one is recusrive !
const Content = ({ data, depth }) => {
  const { sections = [], articles = [] } = data;
  const contents = sections.concat(articles);
  contents.sort(sortByIntOrdre);
  return (
    <>
      {contents.map((content, index) => {
        // if node has a content key, it can only be an article
        if (content.content) {
          return <Article key={index} {...content} />;
        }
        return (
          <React.Fragment key={index}>
            <Title id={content.id} depth={depth} as={depth ? "h4" : "h3"}>
              {content.title}
            </Title>
            <Content data={content} depth={depth + 1} />
          </React.Fragment>
        );
      })}
    </>
  );
};

Content.propTypes = {
  data: PropTypes.shape({
    title: PropTypes.string,
    id: PropTypes.string.isRequired,
    sections: PropTypes.array,
    articles: PropTypes.arrayOf(
      PropTypes.shape({
        content: PropTypes.string
      })
    )
  }).isRequired,
  depth: PropTypes.number.isRequired
};

export default Content;

const { fonts } = theme;

const Title = styled.h3`
  font-size: ${props => (props.depth ? fonts.sizeH4 : fonts.sizeH3)};
`;
