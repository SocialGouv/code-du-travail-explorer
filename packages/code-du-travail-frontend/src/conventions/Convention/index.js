import React from "react";
import PropTypes from "prop-types";
import Info from "./Info";
import Texte from "./Texte";
import Explainer from "./Explainer";
import ListTextes from "./ListTextes";
import { Tabs } from "@cdt/ui-old";

const Convention = ({ convention, container }) => {
  return (
    <>
      <Explainer />
      <Tabs
        data={[
          {
            tab: "Informations",
            panel: <Info convention={convention} container={container} />,
            key: "infos"
          },
          {
            tab: "Texte de base",
            panel: <Texte node={container.content} />,
            key: "texte-de-base"
          },
          {
            tab: "Texte attach√©s",
            panel: (
              <ListTextes
                conventionId={container.conventionId}
                typeTextes={"attaches"}
              />
            ),
            key: "attaches"
          },
          {
            tab: "Texte salaires",
            panel: (
              <ListTextes
                conventionId={container.conventionId}
                typeTextes={"salaires"}
              />
            ),
            key: "salaires"
          }
        ]}
      />
    </>
  );
};

Convention.propTypes = {
  convention: PropTypes.object.isRequired,
  container: PropTypes.shape({
    content: PropTypes.object.isRequired
  }).isRequired
};

export default Convention;
