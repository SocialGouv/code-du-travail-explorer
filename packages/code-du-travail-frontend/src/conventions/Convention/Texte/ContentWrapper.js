import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Content from "./Content";
import { Section, theme } from "@cdt/ui";

const ContentWrapper = ({ data, title }) => {
  return (
    <StyledSection>
      {title && <H2>{title}</H2>}
      <Content depth={0} data={data} />
    </StyledSection>
  );
};

ContentWrapper.propTypes = {
  data: PropTypes.object.isRequired,
  title: PropTypes.string
};

export default ContentWrapper;

const { breakpoints, spacing } = theme;

const StyledSection = styled(Section)`
  @media (min-width: ${breakpoints.tablet}) {
    flex: 0 1 65%;
    overflow: scroll;
  }
`;

const H2 = styled.h2`
  margin-top: 0;
  margin-bottom: ${spacing.large};
`;
