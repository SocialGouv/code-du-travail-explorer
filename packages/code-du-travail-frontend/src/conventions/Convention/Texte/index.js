import React from "react";
import PropTypes from "prop-types";

import { theme } from "@cdt/ui/";
import Sidebar from "./Sidebar";
import ContentWrapper from "./ContentWrapper";
import styled from "styled-components";

const Texte = ({ data, title }) => {
  let rootSection = data;
  // skip the first content level if it's only a "wrapper"
  if (
    rootSection.sections &&
    rootSection.sections.length === 1 &&
    rootSection.sections[0].sections
  ) {
    rootSection = data.sections[0];
  }
  return (
    <Wrapper>
      <Sidebar data={rootSection} />
      <ContentWrapper data={rootSection} title={title} />
    </Wrapper>
  );
};

Texte.propTypes = {
  data: PropTypes.shape({
    title: PropTypes.string,
    sections: PropTypes.array
  }).isRequired
};

export default Texte;

const Wrapper = styled.div`
  @media (min-width: ${theme.breakpoints.tablet}) {
    display: flex;
    justify-content: center;
  }
`;
