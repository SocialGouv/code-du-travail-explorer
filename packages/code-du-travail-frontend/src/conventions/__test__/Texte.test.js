import React from "react";
import { render } from "@testing-library/react";
import Texte from "../Convention/Texte";
import { containerAndTexteDeBase } from "./api.conventions.mock";

describe("<Texte />", () => {
  it("renders without a title", () => {
    const { container } = render(
      <Texte data={containerAndTexteDeBase.texteDeBase} />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders with a title", () => {
    const { container } = render(
      <Texte
        data={containerAndTexteDeBase.texteDeBase}
        title="This title must be in snapshot"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
