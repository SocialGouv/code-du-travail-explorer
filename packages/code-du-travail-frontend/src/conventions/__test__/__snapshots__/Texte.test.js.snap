// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Texte /> renders with a title 1`] = `
.c2 {
  padding: 1.25rem 0;
}

.c4 {
  padding: 0.625rem 1.25rem;
  color: #434956;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  border-color: #c9d3df;
  background-color: #ebeff3;
}

.c4 > *:first-child {
  margin-top: 0;
}

.c4 > *:last-child {
  margin-bottom: 0;
}

.c5 {
  font-size: 0.875rem;
}

.c6 {
  display: block;
  padding: 0.25rem 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:hover {
  font-weight: bold;
}

.c3 {
  max-height: 100%;
  overflow: auto;
}

.c10 {
  margin-bottom: 0.625rem;
  font-style: italic;
  font-weight: bold;
  font-size: 1em;
}

.c11 {
  margin-bottom: 0.625rem;
}

.c9 {
  font-size: 1.625rem;
}

.c8 {
  margin-top: 0;
  margin-bottom: 2rem;
}

@media (max-width:600px) {
  .c4 {
    padding: 0.625rem 1.25rem;
  }
}

@media print {
  .c4 {
    border: none;
    padding: 0 5pt;
  }
}

@media (min-width:980px) {
  .c1 {
    -webkit-flex: 0 1 35%;
    -ms-flex: 0 1 35%;
    flex: 0 1 35%;
    position: -webkit-sticky;
    position: sticky;
    top: 0px;
    max-height: 100vh;
    padding-right: 1.25rem;
  }
}

@media (min-width:980px) {
  .c7 {
    -webkit-flex: 0 1 65%;
    -ms-flex: 0 1 65%;
    flex: 0 1 65%;
    overflow: scroll;
  }
}

@media (min-width:980px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3 c4"
      >
        <ol>
          <li
            class="c5"
          >
            <a
              class="c6"
              href="#KALISCTA000038524044"
            >
              Préambule
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              class="c6"
              href="#KALISCTA000038524045"
            >
              Titre Ier Clauses générales
               
              ▼
            </a>
          </li>
        </ol>
      </div>
    </div>
    <div
      class="c7 c2"
    >
      <h2
        class="c8"
      >
        This title must be in snapshot
      </h2>
      <h3
        class="c9"
        id="KALISCTA000038524044"
      >
        Préambule
      </h3>
      <div
        class="c10"
      >
        Article non étendu
      </div>
      <div
        class="c11"
      >
        <p>
          OK
        </p>
      </div>
      <h3
        class="c9"
        id="KALISCTA000038524045"
      >
        Titre Ier Clauses générales
      </h3>
      <div
        class="c10"
      >
        Article non étendu
      </div>
      <h5
        id="KALIARTI000038524073"
      >
        Commission paritaire nationale de l'emploi
      </h5>
      <div
        class="c11"
      >
        <p>
          OK
        </p>
      </div>
      <div
        class="c10"
      >
        Article non étendu
      </div>
      <div
        class="c11"
      >
        <p>
          OK
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Texte /> renders without a title 1`] = `
.c2 {
  padding: 1.25rem 0;
}

.c4 {
  padding: 0.625rem 1.25rem;
  color: #434956;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.25rem;
  border-color: #c9d3df;
  background-color: #ebeff3;
}

.c4 > *:first-child {
  margin-top: 0;
}

.c4 > *:last-child {
  margin-bottom: 0;
}

.c5 {
  font-size: 0.875rem;
}

.c6 {
  display: block;
  padding: 0.25rem 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:hover {
  font-weight: bold;
}

.c3 {
  max-height: 100%;
  overflow: auto;
}

.c9 {
  margin-bottom: 0.625rem;
  font-style: italic;
  font-weight: bold;
  font-size: 1em;
}

.c10 {
  margin-bottom: 0.625rem;
}

.c8 {
  font-size: 1.625rem;
}

@media (max-width:600px) {
  .c4 {
    padding: 0.625rem 1.25rem;
  }
}

@media print {
  .c4 {
    border: none;
    padding: 0 5pt;
  }
}

@media (min-width:980px) {
  .c1 {
    -webkit-flex: 0 1 35%;
    -ms-flex: 0 1 35%;
    flex: 0 1 35%;
    position: -webkit-sticky;
    position: sticky;
    top: 0px;
    max-height: 100vh;
    padding-right: 1.25rem;
  }
}

@media (min-width:980px) {
  .c7 {
    -webkit-flex: 0 1 65%;
    -ms-flex: 0 1 65%;
    flex: 0 1 65%;
    overflow: scroll;
  }
}

@media (min-width:980px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3 c4"
      >
        <ol>
          <li
            class="c5"
          >
            <a
              class="c6"
              href="#KALISCTA000038524044"
            >
              Préambule
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              class="c6"
              href="#KALISCTA000038524045"
            >
              Titre Ier Clauses générales
               
              ▼
            </a>
          </li>
        </ol>
      </div>
    </div>
    <div
      class="c7 c2"
    >
      <h3
        class="c8"
        id="KALISCTA000038524044"
      >
        Préambule
      </h3>
      <div
        class="c9"
      >
        Article non étendu
      </div>
      <div
        class="c10"
      >
        <p>
          OK
        </p>
      </div>
      <h3
        class="c8"
        id="KALISCTA000038524045"
      >
        Titre Ier Clauses générales
      </h3>
      <div
        class="c9"
      >
        Article non étendu
      </div>
      <h5
        id="KALIARTI000038524073"
      >
        Commission paritaire nationale de l'emploi
      </h5>
      <div
        class="c10"
      >
        <p>
          OK
        </p>
      </div>
      <div
        class="c9"
      >
        Article non étendu
      </div>
      <div
        class="c10"
      >
        <p>
          OK
        </p>
      </div>
    </div>
  </div>
</div>
`;
